### PR TITLE
Job Builder Skills Component

### LINKED ISSUE #967: Job Poster Builder: Step 06 - Skills
# Description
The sixth step in creating a job poster is Skills. A skill is the ability (or expertise) to perform a specific function in order to complete a Key Task (or Tasks). Skills and their definitions are provided by Talent Cloud and are selected by the user based on the Key Tasks previously written by the user in Job Poster Builder: Step 05 - Tasks #959 

- [x] This task requires a unit test.
- [x] This task is complex and needs acceptance criteria.

**Acceptance Criteria**

***Progress Bar***
- [x] As skills are added they get added to either one or both of the progress bar counters. One bar indicated the number of Essential Skills that have been added and the other indicates the number of total skills added.
- [x] Each progress bar will indicate a sweet spot that is an indication of optimized for best results. 

***Skills, Work Environment***
- [ ] On page load a minimum of 1 and a maximum of 2 Work Environment skills will be presented to the user within the Work Environment section of this page. These pre-selected skills will be created based on the results within Job Poster Builder: Step 03 - Work Environment #745.
- [x] Users will have the ability to add the the recommended skills or add any other skill listed.

***Skills***
- [x] On page load the user will be presented with the list of Key Tasks as previously written by the user in Job Poster Builder: Step 05 - Tasks #959.  
- [x] Skills are categorized as Occupational Skills, Cultural Skills and Future Skills.
- [x] All skills will be manually tagged to fall into one (or more) of these categories.
- [x] Once a skill is added a modal will pop-up allowing the user to add specificity to the definition and to determine the level.
- [ ] The definition will be capped at 50 words.
- [x] Once a skill has been added, it will be added to the list of skills under either Essential or Asset.
- [x] Once a skill has been added, the Skills counter will increase by a value of one. If the Skill is Essential, the Essential Skills counter will also increase by one. 
- [x] Skills can be removed by either the selecting the icon on the skill itself or from the icon within the skills list. 
- [x] Skills can be edited by selecting the icon within the skills list.
- [x] Users will have the ability to Save their progress through a save button within the skills list area. 
- [x] If the user wants a new skill added, they will have an option at the bottom of the page to Add Your Own Skill. This will be an email to the Talent Cloud team to inquiry about adding a new skill. 

***Saving***
- [x] Once a level has been selected the user will **Save and Preview**. This will take the user to a new page highlighting the Key Task, the Skill, the definition and the level. 

# Link to wireframes:
[Skills wireframe](https://xd.adobe.com/view/b27b7bbc-5560-4e73-65aa-1c12787a2c94-c44a/?fullscreen&hints=off)

# Link to content:
[Skills content](https://docs.google.com/document/d/16Yqg5Q8Sruk1ujdaq8nkYMFXz2OReHqLf-8PSWczNzc/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #967 .

### Notes:
- Up/down buttons to reorder skills should be added in later.
- Recommended Culture skills must be added in once the business logic is decided.
- Should add a 4th section of skills, containing anything missing from the other sets.
- ~The textarea on the CriteriaForm needs to change, but it requires a change to the backend as well. Instead of just one field (description) it should save two: description and specificity.~ Edit: fixed with c2a4cb67468417f4cc18fda8ea4df26bddb7626d

### CODE REVIEW

File: resources/assets/js/components/JobBuilderSkills/CriteriaForm.tsx [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+                            trigger
+                          />
+                        );
+                      },
+                    )}
+                    <div
+                      className="job-builder-skill-level-or-block"
+                      data-c-alignment="base(centre)"
+                    >
+                      {/** This empty div is required for CSS magic */}
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/components/JobBuilderSkills/CriteriaForm.tsx]: 😢

File: resources/assets/js/fakeData/fakeSkills.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+export const fakeSkill = (): Skill => ({
+  id: 1,
+  skill_type_id: SkillTypeId.Hard,
+  en: {
+    name: "Front-end development",
+    description:
+      "Defined as: Developing web applications using HTML5, CSS3, Javascript",
+  },
+  fr: {
+    name: "Dé9veloppement frontal",
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/fakeData/fakeSkills.ts]: yep

File: resources/assets/js/components/JobBuilderSkills/JobBuilderSkills.tsx [hunk 3]
```diff
@@ -0,0 +1,1284 @@
...
+            >
+              Save &amp; Preview Skills
+            </button>
+          </div>
+        </div>
+      </div>
+      <div data-c-dialog-overlay={isModalVisible ? "active" : ""} />
+      {/** This modal is for adding brand new skills */}
+      <Modal
+        id="job-bulder-add-skill"
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/components/JobBuilderSkills/JobBuilderSkills.tsx]: Makes sense.

[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/components/JobBuilderSkills/JobBuilderSkills.tsx]: Actually... are the modal ids even necessary? Josh uses them to link modals to buttons, but we're bypassing all that anyway. Just seems like something else we need to track for no reason.

[the remark this answers is not in the document]
Comment (Khristophor) [reply resources/assets/js/components/JobBuilderSkills/JobBuilderSkills.tsx]: The only other thing is for accessibility, the ID gets passed to `aria-title` and `aria-described-by` that correspond to `modal-id-title` and `modal-id-description` respectively

File: resources/assets/js/api/skill.ts [hunk 4]
```diff
@@ -11,20 +11,15 @@ export const parseSkillTranslation = (
...
 });
 
-export const parseSkill = (data: ResponseData): Skill => ({
-  id: Number(data.id),
-  skill_type_id: Number(data.skill_type_id),
-  // TODO: remove the hasKey hack when removing skill from Criteria
-  en: parseSkillTranslation(data.en),
-  fr: parseSkillTranslation(data.fr),
-});
+export const parseSkill = (data: any): Skill => data;
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/assets/js/api/skill.ts]: We don't need it, but I like keeping it around for now as a reminder that it should be _better_ in the future.

**Pull request merged**